app/routes.py

```diff
@@ -1,2 +1,187 @@
 from flask import Blueprint
+from app import db
+from .models.task import Task
+from .models.goal import Goal
+from flask import request, jsonify, make_response
+from datetime import datetime
+import requests
+from app import slack_key
 
+task_list_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goal_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@task_list_bp.route("", methods = ["POST"], strict_slashes = False)
+def create_task():
+    """fdfd"""
+    request_body = request.get_json()
+
+    if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+        return make_response(jsonify({"details": f"Invalid data" }), 400)
+    
+    new_task = Task(title=request_body["title"],
+                    description=request_body["description"],
+                    completed_at=request_body["completed_at"],
+                    goal_id=request_body["goal_id"] if request_body.get("goal_id") else None) #Wave6 one to many relation
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return make_response({
+        "task": new_task.to_json()}, 201)
+
+@task_list_bp.route("", methods = ["GET"])
+def get_all_tasks():
+    if request.args.get("sort") == "asc":
+        tasks = Task.query.order_by(Task.title)
+    elif request.args.get("sort") == "desc":
+        tasks = Task.query.order_by(Task.title.desc())
+    elif request.args.get("sort") == "id":
+        tasks = Task.query.order_by(Task.task_id)
+    elif request.args.get("title"):
+        tasks = Task.query.order_by(Task.task_id)
+    else:
+        tasks = Task.query.all()
+
+    task_list = []
+    for task in tasks:
+        task_list.append(task.to_json())
+    return make_response(jsonify(task_list))
+
+@task_list_bp.route("/<task_id>", methods = ["GET", "PUT", "DELETE"])
```

Comment:
For the route "" the "GET" and "POST" methods are split into different functions, while multiple methods are combined for this route. Either way is fine, but I recommend using a consistent style throughout.